src/Client.php

```diff
@@ -182,6 +182,13 @@ public function send(RequestInterface $request)
                 return new FutureResponse(new RejectedPromise($e));
             }
             throw RequestException::wrapException($trans->request, $e);
+        } catch (\TypeError $error) {
```

Comment:
Why just `TypeError` and not `Error` or `Throwable`. Haven't looked to deep, so you might have better insights ATM.